rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -473,6 +491,12 @@ func (r *Layer2Relayer) ProcessPendingBatches() {
 		log.Info("Forcing submission of batches due to timeout", "batch index", batchesToSubmit[0].Batch.Index, "first block created at", oldestBlockTimestamp)
 	}
 
+	// Sanity checks before constructing the transaction
+	if err := r.sanityChecksBeforeConstructingTransaction(batchesToSubmit); err != nil {
```

Comment:
Some of these checks seem the same (or at least very similar) as they should be done when constructing the `batchesToSubmit`. 

I'm thinking whether it would be more meaningful to do these checks based on the constructed transaction. ie we decode the batch header from calldata and everything else from the blobs and then try to make sense of the decoded data (and relate it back to the DB). This way we would catch any error in constructing the batches and encoding.

Reply:
Added in commit: [925ef74](https://github.com/scroll-tech/scroll/pull/1714/commits/925ef74867c4a2f3beca276aa3b0c6c834ff4784).

Reply:
Refactored in [cf85225](https://github.com/scroll-tech/scroll/pull/1714/commits/cf852259f36191d55e7d92a77b8fd8a6523d667d).